result/result.py

```diff
@@ -115,6 +116,42 @@ def unwrap_or(self, default: T) -> T:
         else:
             return default
 
+    def map(self, op: Callable[[T], U]) -> 'Result[E, U]':
+        """
+        Return `Ok` with original value mapped to a new value using the passed
+        in function. Otherwise return `Err` with same value.
+        """
+        if self._is_ok:
+            return Ok(op(cast(T, self._value)))
+        else:
+            return Err(cast(E, self._value))
```

Comment:
Hm, in this case, couldn't we just `return self`? After all, the error value will be unchanged. The same counts for the other two map functions.

Reply:
Yeah. I've thought about this, but wasn't sure if it was proper to return the same value due to aliasing. But I suppose if you're just comparing two Maybe's using their API instead of in any other way, it's fine and a good optimization.

I'll make the change though.

Reply:
Aliasing would be perfectly fine, after all `map` will leave a non-`Err`-value untouched.